### ISSUE TITLE
gomobile building for Android

### ISSUE THREAD

**Issue by n8fr8** [ipfs/kubo#2816]
I was able to build ipfs using the latest go 1.6.2 and gomobile tools, to create both an APK and an AAR library file. Yay!

The one issue is that it didn't like the fact the package was named "main" since that is a bit of a reserved word. Any chance that could be changed to something like main-ipfs?

**Comment by whyrusleeping** [comment]
Packages in go must be named main in order for a binary to be created

**Comment by n8fr8** [comment]
Right, so with Android we aren't creating a binary of that sort, so we'll have to do something else.
